bundles/org.openhab.binding.yamahamusiccast/src/main/java/org/openhab/binding/yamahamusiccast/internal/YamahaMusiccastHandler.java

```diff
@@ -0,0 +1,1331 @@
+                    setVolume(newVolume, "zone2", host);
+                    break;
+                case 3:
+                    tmpString = getStatus(host, "zone3");
+                    targetObject = new Gson().fromJson(tmpString, Status.class);
+                    responseCode = targetObject.getResponseCode();
+                    maxVolumeLinkedDevice = targetObject.getMaxVolume();
+                    newVolume = maxVolumeLinkedDevice * value / 100;
+                    setVolume(newVolume, "zone3", host);
+                    break;
+                case 4:
+                    tmpString = getStatus(host, "zone4");
+                    targetObject = new Gson().fromJson(tmpString, Status.class);
+                    responseCode = targetObject.getResponseCode();
+                    maxVolumeLinkedDevice = targetObject.getMaxVolume();
+                    newVolume = maxVolumeLinkedDevice * value / 100;
+                    setVolume(newVolume, "zone4", host);
+                    break;
+            }
+        }
+    }
+
+    private void updateMCLinkStatus() {
+        tmpString = getDistributionInfo(this.host);
+        @Nullable
+        DistributionInfo targetObject = new Gson().fromJson(tmpString, DistributionInfo.class);
+        role = targetObject.getRole();
+        ChannelUID testchannel;
+        if (role.equals("none")) {
+            for (int i = 1; i <= zoneNum; i++) {
+                switch (i) {
+                    case 1:
+                        testchannel = new ChannelUID(getThing().getUID(), "main", "channelMCServer");
+                        if (isLinked(testchannel)) {
+                            updateState(testchannel, StringType.valueOf(""));
+                        }
+                        break;
+                    case 2:
+                        testchannel = new ChannelUID(getThing().getUID(), "zone2", "channelMCServer");
+                        if (isLinked(testchannel)) {
+                            updateState(testchannel, StringType.valueOf(""));
+                        }
+                        break;
+                    case 3:
+                        testchannel = new ChannelUID(getThing().getUID(), "zone3", "channelMCServer");
+                        if (isLinked(testchannel)) {
+                            updateState(testchannel, StringType.valueOf(""));
+                        }
+                        break;
+                    case 4:
+                        testchannel = new ChannelUID(getThing().getUID(), "zone4", "channelMCServer");
+                        if (isLinked(testchannel)) {
+                            updateState(testchannel, StringType.valueOf(""));
+                        }
+                        break;
+                }
+            }
+        }
+    }
+    // End Various functions
+
+    // API calls to AVR
+
+    // Start Zone Related
+
+    private String getStatus(String host, String zone) {
```

Comment:
These methods have much code in common. You could add a method which takes the last part of the URL as an argument. This method can be invoked by the methods below, then.

Reply:
Prepared changes, will be available in next push